bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/service/ConfigurableServiceResource.java

```diff
@@ -0,0 +1,220 @@
+ *
+ * @author Dennis Nobel - Initial contribution
+ *
+ */
+@Path(ConfigurableServiceResource.PATH_SERVICES)
+@Api
+public class ConfigurableServiceResource implements RESTResource {
+
+    /** The URI path to this resource */
+    public static final String PATH_SERVICES = "services";
+
+    private static final String CONFIGURABLE_SERVICE_FILTER = "(" + ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI
+            + "=*)";
+
+    private final Logger logger = LoggerFactory.getLogger(ConfigurableServiceResource.class);
+
+    private ConfigurationAdmin configurationAdmin;
+
+    @GET
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Get all configurable services.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK") })
+    public List<ConfigurableServiceDTO> getAll() {
+        List<ConfigurableServiceDTO> services = getConfigurableServices();
+        return services;
+    }
+
+    @GET
+    @Path("/{serviceId}")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Get configurable service for given service ID.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"), @ApiResponse(code = 404, message = "Not found") })
+    public Response getById(@PathParam("serviceId") @ApiParam(value = "service ID", required = true) String serviceId) {
+        List<ConfigurableServiceDTO> configurableServices = getConfigurableServices();
+        for (ConfigurableServiceDTO configurableService : configurableServices) {
+            if (configurableService.id.equals(serviceId)) {
+                return Response.ok(configurableService).build();
+            }
+        }
+        return Response.status(404).build();
+    }
+
+    @GET
+    @Path("/{serviceId}/config")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Get service configuration for given service ID.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"), @ApiResponse(code = 404, message = "Not found"),
+            @ApiResponse(code = 500, message = "Configuration can not be read due to internal error") })
+    public Response getConfiguration(
+            @PathParam("serviceId") @ApiParam(value = "service ID", required = true) String serviceId) {
+        try {
+            Configuration servieConfiguration = configurationAdmin.getConfiguration(serviceId);
+            Dictionary<String, Object> properties = servieConfiguration.getProperties();
+            Map<String, Object> configuration = toMap(properties);
+            return configuration != null ? Response.ok(configuration).build() : Response.status(404).build();
+        } catch (IOException ex) {
+            logger.error("Cannot get configuration for service {}: " + ex.getMessage(), serviceId, ex);
+            return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @POST
```

Comment:
just noticed that you use POST here and not PUT. In `/things/{thingUID}/config` we are using PUT - I think both should be the same. As we are replacing an existing config here, my personal preference would be PUT.